frontera/contrib/scrapy/schedulers/frontier.py

```diff
@@ -124,11 +134,13 @@ def process_exception(self, request, exception, spider):
     def open(self, spider):
         log.msg('Starting frontier', log.INFO)
         if not self.frontier.manager.auto_start:
-            self.frontier.start()
+            scrapy_kwargs = {'spider': spider}
+            self.frontier.start(**scrapy_kwargs)
```

Comment:
I think that understanding how hcf backend works, some of the features may have sense (for example, the spider reference one). Check code:

https://github.com/scrapinghub/hcf-backend/blob/master/hcf_backend/backend.py